File: CPP/cpp 2025/multi_table.cpp
#include <iostream>
using namespace std;

// Made in 03/12/2025, re-made in 07/12/2025

int mainCode();

void multi_table(int number, int start, int end){
    for(start; start <= end; start++){
        cout << number << " x " << start << " = " << number*start << endl;
    }
}

int main(){
    cout << "Digit 3 numbers to make a multiplication table! :)\n";
    cout << "Digit 0 in all 3 to finish the program.\n\n";

    int again = mainCode();
    while (again != 0){
        again = mainCode();
    }
    cout << "Finishing...";
}

int mainCode(){
    int n, s, e;
        cout << "The main number: \n"; cin >> n;
    while (true){
        cout << "The start: \n"; cin >> s;
        cout << "The end: \n"; cin >> e;

        if (n == 0 && s == 0 && e == 0){return 0;}
        else if (s > e){cout << "Sorry, end needs to be higher than start.\n"; continue;}
        else{break;}
    }
    multi_table(n, s, e);
    return 1;
}

File: Projects/Other/Multiplication table/multi_table.py
# Code made in 19/01/2026

# Input, validation and calculates the mult table
# Returns true or false (true: code is repeated, false: program finishes)
def run_mult_table():
    #inputs
    index = int(input("Index: "))
    if index == 0:
        return False
    
    start = int(input("Start: "))
    end = int(input("End: "))

    while start > end: # validation
        print("Start must be smaller than end.")
        start = int(input("Start: "))
        end = int(input("End: "))

    # calculating and printing the result
    while start <= end:
        print(f"{index} x {start} = {index*start}")
        start += 1
    return True



def main(): # main code
    print("\nDigit 3 numbers to calculate the multiplication table!")
    print("Digit 0 on the index number to finish the program.\n")

    again = run_mult_table() 
    while again: #loops until return 0
        again = run_mult_table()

main()